pkg/ssh/keymanager.go

```diff
@@ -0,0 +1,301 @@
+package ssh
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/rand"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/grafana/dskit/services"
+	"github.com/grafana/pdc-agent/pkg/pdc"
+	"github.com/mikesmitty/edkey"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// SSHKeySize is the size of the SSH key.
+	SSHKeySize     = 4096
+	KnownHostsFile = "pdc_known_hosts"
+)
+
+// KeyManager manages SSH keys and certificates. It ensures that the SSH keys,
+// certificates and known_hosts files exist in their configured locations.
+// EnsureCertExists should only be called once the service has started successfully.
+
+type KeyManager interface {
+	services.Service
+	EnsureCertExists(bool) error
+}
+
+// pdcKeyManager implements KeyManager. If needed, it gets new certificates signed
+// by the PDC API.
+//
+// If the service starts successfully, then the key and cert files will exist.
+// It will attempt to reuse existing keys and certs if they exist.
+type pdcKeyManager struct {
+	*services.BasicService
+	cfg    *Config
+	client pdc.Client
+	logger log.Logger
+}
+
+// NewKeyManager returns a new KeyManager in an idle state
+func NewKeyManager(cfg *Config, logger log.Logger, client pdc.Client) KeyManager {
+	km := pdcKeyManager{
+		cfg:    cfg,
+		client: client,
+		logger: logger,
+	}
+
+	km.BasicService = services.NewIdleService(km.starting, km.stopping)
+	return &km
+}
+
+func (km *pdcKeyManager) starting(_ context.Context) error {
+	level.Info(km.logger).Log("msg", "starting key manager")
+
+	newCertRequired := km.cfg.ForceKeyFileOverwrite
+	var err error
+
+	if !newCertRequired {
+		newCertRequired, err = km.ensureKeysExist()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Slightly different behaviour than the one documented. 

The documentation mentions:
```If enabled, the pdc-agent will regenerate an SSH key pair and request a new certificate to use when establishing an SSH tunnel.```

The code never re-generate the SSH key pair !

How about removing the flag for now ? User can simply delete the key-pair if needed ?

Reply:
Removing it seems much nicer :)